@xen-orchestra/vmware-explorer/esxi.mjs

```diff
@@ -25,8 +26,10 @@ export default class Esxi extends EventEmitter {
     //
     // we need to find a fix for this, maybe forking the library
     this.#client = new Client(host, user, password, sslVerify)
-    this.#client.once('ready', () => {
+    this.#client.once('ready', async () => {
       this.#ready = true
+      const res = await this.search('Datacenter', ['name'])
```

Comment:
You need to handle any potential errors.

`EventEmitter` is not promise-aware, it will not catch a rejection which will lead to the `unhandledRejection` error.